mkosi/__init__.py

```diff
@@ -176,6 +176,41 @@ def remove_packages(state: MkosiState) -> None:
             die(f"Removing packages is not supported for {state.config.distribution}")
 
 
+def configure_os_release(state: MkosiState) -> None:
+    """Write IMAGE_ID and IMAGE_VERSION to /usr/lib/os-release in the image."""
+    if not state.config.image_id and not state.config.image_version:
+        return
+
+    for candidate in ["usr/lib/os-release", "etc/os-release", "usr/lib/initrd-release", "etc/initrd-release"]:
+        osrelease = state.root / candidate
+        # at this point we know we will either change or add to the file
+        newosrelease = osrelease.with_suffix(".new")
+
+        if not osrelease.is_file():
```

Comment:
We should probably skip symlinks here

Reply:
Right, only checked *dangling* symlinks there.